dao_data_model.md

```diff
@@ -0,0 +1,227 @@
+  pub governance_modules: UnorderedSet<AccountId>,
+  /// Extensions keep track of registry of DAO approved extensions (keys), where state requirements are wrapped in value - if specific data needs to remain within the DAO vs inside the extension. (Think of DAO preferences or pagination indexes)
+  pub extension_modules: UnorderedMap<AccountId, Vec<u8>>,
+  
+  /// NOTE: Policy may need changes to support diff groups control over governance modules/extensions
+}
+```
+
+
+## DAO Proposal Data
+NOTE: this would utilize the "VersionedProposal" setup, adding a new version
+
+NOTE: Need to consider items brought up in: https://github.com/near-daos/sputnik-dao-contract/issues/161
+
+```rust
+pub struct Proposal {
+// ---- ORIGINAL UNCHANGING ----------------------------
+    /// Original proposer.
+    pub proposer: AccountId,
+    /// Description of this proposal.
+    pub description: String,
+    /// Kind of proposal with relevant information.
+    pub kind: ProposalKind,
+    /// Current status of the proposal.
+    pub status: ProposalStatus,
+    /// Submission time (for voting period).
+    pub submission_time: U64,
+// ---- CHANGING PROPOSAL ----------------------------
+    /// DEPRECATE: Count of votes per role per decision: yes / no / spam. (Can be safely deprecated upon all active proposals being finalized)
+    pub vote_counts: HashMap<String, [Balance; 3]>,
+    /// DEPRECATE: Map of who voted and how. (Can be safely deprecated upon all active proposals being finalized)
+    pub votes: HashMap<AccountId, Vote>,
+// ---- NEW PROPOSAL DATA ----------------------------
+    /// The module used to compute data against
+    pub governance_module: AccountId,
+    /// Proofs are votes with a more discrete / flexible payload the math modules can utilize
+    /// NOTE: Map allows for ballot edits up until proposal finalization time
+    pub ballots: UnorderedMap<AccountId, Ballot>,
+    /// Computed index-based tally totals
+    pub outcome: Outcome,
+    /// metadata, allowing extensions or other innovations
+    /// NOTE: This should allow URIs to IPFS docs or other proposal data needed for governance decisions
+    pub metadata: Option<Vec<u8>>,
+}
+```
+
+
+## New Proposal Kinds
+```rust
+pub enum ProposalKind {
+    /// Enable conviction weight decisions to trigger differing functions/actions, 
+    /// This will still only execute 1 proposal kind, however it allows the members 
+    /// to decide the path based on multiple choice (or similar)
+    IndexTriggeredActions(Vec<ProposalKind>),
+}
+```
+
+
+## Governance Module Interface
+```rust
+// no state please, math only
+// no owners allowed, confirm no access keys before use
+pub trait Governance {
+  /// Implements a function that call tally a set of data, returning the collated value
+  pub fn compute(&self, ballots: Base64VecU8) -> PromiseOrValue<Outcome>;
```

Comment:
Agree with this interface, but Outcome can be replace by the existing `ProposalStatus`.